### PR TITLE
Tf upgrade to 0.12

### PR DESCRIPTION
**Pull request by minniux**
Add support for Terraform 0.12.x

### CODE REVIEW

File: redis/main.tf [hunk 1]
```diff
@@ -1,60 +1,61 @@
...
+  tags = {
     Name        = "${var.project}-${var.environment}-${var.name}"
-    Environment = "${var.environment}"
-    Project     = "${var.name}"
+    Environment = var.environment
+    Project     = var.name
   }
 }
 
 resource "aws_elasticache_replication_group" "redis_auth" {
```
Comment (duboisph) [inline redis/main.tf]: You should be able to simplify the module now by using `null` as default for the var.auth_token, removing the need for the separate redis_auth resource.

File: redis-snapshot-replicator/main.tf [hunk 2]
```diff
@@ -219,145 +224,149 @@ data "archive_file" "create_zip" {
 
 locals {
   # Solution from this comment to open issue on non-relative paths  # https://github.com/hashicorp/terraform/issues/8204#issuecomment-332239294
-  count    = "${var.enable ? 1 : 0}"
-  filename = "${substr(data.archive_file.create_zip.output_path, length(path.cwd) + 1, -1)}"
-
+  count = var.enable ? 1 : 0
+  filename = substr(
```
Comment (iuriaranda) [inline redis-snapshot-replicator/main.tf]: I think this is not really needed anymore, the issue described above is resolved in TF 0.12 afaik.

File: redis/variables.tf [hunk 3]
```diff
@@ -1,88 +1,104 @@
...
 variable "engine_version" {
   description = "The redis engine version"
   default     = "3.2.6"
+  type        = string
 }
 
 variable "port" {
   description = "The redis port"
   default     = "6379"
+  type        = string
```
Comment (iuriaranda) [inline redis/variables.tf]: This should be type `number`

File: redis/variables.tf [hunk 4]
```diff
@@ -1,88 +1,104 @@
...
 variable "snapshot_window" {
   description = "The daily time range (in UTC) during which ElastiCache will begin taking a daily snapshot of your cache cluster. The minimum maintenance window is a 60 minute period. Example: 05:00-09:00"
   default     = "03:00-05:00"
+  type        = string
 }
 
 variable "snapshot_retention_limit" {
   description = "The number of days for which ElastiCache will retain automatic cache cluster snapshots before deleting them. For example, if you set SnapshotRetentionLimit to 5, then a snapshot that was taken today will be retained for 5 days before being deleted. If the value of SnapshotRetentionLimit is set to zero (0), backups are turned off. Please note that setting a snapshot_retention_limit is not supported on cache.t1.micro or cache.t2.* cache nodes"
   default     = "0"
+  type        = number
```
Comment (iuriaranda) [inline redis/variables.tf:78]: Default should be a number if type is `number`

**Review by iuriaranda — APPROVED** [review]
Some minor things, other than that LGTM

File: redis/main.tf (hunk 1 above)
Comment (iuriaranda) [reply redis/main.tf]: There'll be a terraform state move needed for current setups that use auth. I'd make sure to document that in the release notes

File: redis-snapshot-replicator/main.tf [hunk 5]
```diff
@@ -241,8 +231,8 @@ resource "aws_lambda_function" "redis_create_snapshot" {
   role          = aws_iam_role.iam_for_lambda_redis[0].arn
   handler       = "create_snapshot.lambda_handler"
 
-  filename         = local.filename
-  source_code_hash = filebase64sha256(local.filename)
+  filename         = "${path.module}/shipper.zip"
+  source_code_hash = filebase64sha256("${path.module}/shipper.zip")
```
Comment (iuriaranda) [inline redis-snapshot-replicator/main.tf]: The `archive_file` data source already outputs the base64 encoded sha256 of the archive: https://www.terraform.io/docs/providers/archive/d/archive_file.html#output_base64sha256

So you don't need to re-compute it here.

Comment (minniux) [reply redis-snapshot-replicator/main.tf]: nice! didn't know that. Going to use it

**Review by iuriaranda — APPROVED** [review]

**Review by duboisph — APPROVED** [review]

**Pull request merged**